### PR TITLE
small fixes

### PR DESCRIPTION
**Pull request by sanderegg**
- only create an internal network if an interactive service is made of more than 1 docker container
- fixed some project dependencies
- automatically set the right url in the client side when opening the viewer (osparc01 was hard-coded till now)

### CODE REVIEW

File: services/web-client/source/class/qxapp/components/workbench/Workbench.js [hunk 1]
```diff
@@ -327,7 +328,7 @@ qx.Class.define("qxapp.components.workbench.Workbench", {
           "value": 0
         }],
         "viewer": {
-          "ip": "http://osparc01.speag.com",
+          "ip": "http://" + window.location.hostname,
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/web-client/source/class/qxapp/components/workbench/Workbench.js]: renamed ip to hostname

File: services/director/producer.py [hunk 2]
```diff
@@ -133,8 +133,9 @@ def start_service(service_name, service_tag, service_uuid):
     check_service_uuid_available(dockerClient, service_uuid)
     login_docker_registry(dockerClient)
 
-    # create a new network to connect the differnt containers
-    docker_network = create_overlay_network_in_swarm(dockerClient, service_name, service_uuid)
+    if len(listOfImages) > 1:
+      # create a new network to connect the differnt containers
+      docker_network = create_overlay_network_in_swarm(dockerClient, service_name, service_uuid)
```
Comment (pcrespov) [inline services/director/producer.py]: I am surprised to see that python linter is passing! ``dockerClient, listOfNames, dockerImagePath``... are non compliant. 
Is pylint actually running in travis?

Comment (odeimaiz) [reply services/director/producer.py]: Definitely running "pylint ./services" using .pylintrc rules.
I will check if it's actually really checking all py files

Comment (pcrespov) [reply services/director/producer.py]: Yeah, I saw the log :) but it does not seem to be running

Comment (sanderegg) [reply services/director/producer.py]: pylint does not seem to work.
I renamed all the variables i could find that were camelcase.

Comment (pcrespov) [reply services/director/producer.py]: i will check pylint configuration  ...

**Pull request merged**